community/blog/2021-07 The power of Arcus secret store/README.md

```diff
@@ -0,0 +1,149 @@
+The secret store also allows you to define your own caching strategy for [custom secret providers](https://security.arcus-azure.net/features/secret-store/create-new-secret-provider) or extending existing ones.
+
+The caching functionality acts as a 'wrapper' arround existing secret providers. Any secret provider you can think of can be 'wrapped' into a cached version.
+
+```csharp
+using Arcus.Security.Core.Caching;
+using Arcus.Security.Core.Caching.Configuration;
+using Arcus.Security.Core.Providers;
+
+.ConfigureSecretStore((config, stores) =>
+{
+    // Configuration cache (optional: default is 5 min).
+    var cacheConfiguration = new CacheConfiguration(TimeSpan.FromMinutes(10));
+
+    // Built-in secret providers.
+    var environmentVariables = new EnvironmentVariableSecretProvider();
+    var cachedEnvironmentVariables = new CachedSecretProvider(environmentVariables, cacheConfiguration);
+
+    // Custom secret providers.
+    var windowsRegistry = new WindowsRegistrySecretProvider();
+    var cachedWindowsRegistry = new CahcedSecretProvider(windowsRegistry, CacheConfiguration.Default);
+});
+```
+
+See also our [asynchrounous cache invalidation](https://background-jobs.arcus-azure.net/features/security/auto-invalidate-secrets) to automatically invalidate cached Azure Key Vault secrets. 
+
+## Plug & play
+
+Our secret store is very easily to set up. All it requires is a single call to register the store `ConfigureSecretStore`, after that you can immediately use our `ISecretProvider` interface abstraction throughout your application.
+
+```csharp
+using Arcus.Security.Core;
+
+[ApiController]
+public class OrderController : ControllerBase
+{
+    private readonly ISecretProvider _secretStore;
+
+    public OrderController(ISecretProvider secretStore)
+    {
+        _secretStore = secretStore;
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> Orders()
+    {
+        string secret = await _secretStore.GetRawSecretAsync("MySecret");
+    }
+}
+```
+
+We are very flexible in the registration of the secret providers. We allow all kinds of combinations and even multiple registrations of the same type. Imagine two Azure Key Vaults with public and user secrets.
+
+```csharp
+using Microsoft.Extensions.Hosting;
+
+.ConfigureSecretStore((config, stores) =>
+{
+    stores.AddAzureKeyVaultWithManagedServiceIdentity("https://public-secrets.vault.azure.net")
+          .AddAzureKeyVaultWithManagedServiceIdentity("https://user-secrets.vault.azure.net"); 
+})
+```
+
+> Note that with our [named secret providers feature](https://security.arcus-azure.net/features/secret-store/named-secret-providers), you can retrieve a single secret provider instead of the combination. This can be useful in a scenario like this.
+
+## Designed for security
```

Comment:
Let's move this to the top, this is the trigger for building the secret store

Reply:
Done that. Thanks! I agree.